src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,31 @@
+import Popup from "./Popup";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor({ popupSelector, handleFormSubmit }) {
+    super(popupSelector);
+    this._handlFormSubmit = handleFormSubmit;
+    this._modalForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
+  }
+
+  renderLoading(isLoading, loadingText = "Saving...") {
```

Comment:
you've implemented the logic for rendering loading via the class methods. That's the best way!